R/stats-mlm-tidiers.R

```diff
@@ -0,0 +1,88 @@
+#' @templateVar class lm
+#' @template title_desc_tidy
+#'
+#' @param x An `lm` object created by [stats::lm()].
+#' @template param_confint
+#' @template param_quick
+#' @template param_unused_dots
+#'
+#' @evalRd return_tidy(regression = TRUE)
+#'
+#' @details If the linear model is an `mlm` object (multiple linear model),
+#'   there is an additional column `response`.
+#'
+#'   If you have missing values in your model data, you may need to refit
+#'   the model with `na.action = na.exclude`.
+#'
+#' @examples
+#'
+#' mod <- lm(cbind(mpg, disp) ~ wt, mtcars)
+#' tidy(mod, conf.int = TRUE)
+#' @aliases lm_tidiers
+#' @export
+#' @seealso [tidy()]
+#' @family lm tidiers
+#' @rdname tidy.mlm
+#' @export
+
+tidy.mlm <- function(x,
+                     conf.int = FALSE,
+                     conf.level = .95,
+                     quick = FALSE,
+                     ...) {
+  # if only term and estimate columns are needed
+  if (quick) {
+    co <- stats::coef(x)
+
+    ret <-
+      data.frame(
+        response = rep(colnames(co), each = nrow(co)),
+        term = rep(rownames(co), times = ncol(co)),
+        estimate = as.numeric(co),
+        stringsAsFactors = FALSE
+      )
+
+    return(process_mlm(ret, 
+                      x, 
+                      conf.int = FALSE))
+  }
+
+  # adding other details from summary object
+  s <- summary(x)
+  ret <- tidy.summary.lm(s)
```

Comment:
`tidy.summary.lm()` shouldn't exist anymore (?) so you may need to recreate some of that processing here. The code duplication is okay because it'll make things more maintainable / decoupled in the long run.

Reply:
I have modified `tidy.summary.lm` to work only with `lm` objects and added a new `tidy.summary.mlm` method for `mlm` objects.